src/v2/expressions.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * A CEL expression which can be evaulated during function deployment, and
+ * resolved to a value of the generic type parameter: i.e, you can pass
+ * an Expression<number> as the value of an option that normally accepts numbers.
+ */
+export type Expression<T extends string | number | boolean> = string; // eslint-disable-line
+
+export type Field<T extends string | number | boolean> =
+  | T
+  | Expression<T>
+  | null;
```

Comment:
We've talked about not using `null` and instead invent `options.ResetToDefault` which is defined to have a `toJSON` which returns `null`